File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
  public static int[] solution(int numer1, int denom1, int numer2, int denom2) {
        // 1. 두 분수의 합을 계산
        int numer = numer1 * denom2 + numer2 * denom1;
        int denom = denom1 * denom2;

        // 2. 분자와 분모의 최대공약수 계산
        int gcd = gcd(numer, denom);

        // 3. 기약 분수로 만들기
        numer /= gcd;
        denom /= gcd;

        return new int[] { numer, denom };
    }

    // 최대공약수 계산 함수 (유클리드 알고리즘)
    public static int gcd(int a, int b) {
        while (b != 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.js
// function solution(numer1, denom1, numer2, denom2) {
//     var answer = [0,0];
//     var Gcd = findGcd(denom1, denom2)
//     answer[0] = (numer1 * denom2 + numer2 * denom1) / Gcd;
//     answer[1] = denom1 * denom2 / Gcd;
//     return answer; 
// }

// function findGcd(a, b) {
// 	const remainder = a % b
//     if (remainder === 0)
//         return b
//     return findGcd(b, remainder);
// }


function solution(numer1, denom1, numer2, denom2) {
    let numer = numer1 * denom2 + numer2 * denom1;
    let denom = denom1 * denom2;

    let gcd = gcdFunction(numer, denom);

    numer /= gcd;
    denom /= gcd;

    return [numer, denom];
}

function gcdFunction(a, b) {
    while (b !== 0) {
        let temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}
